### PR TITLE
Add AIX support 

### PR DESCRIPTION
**Pull request by Helflym**
This Pull Request adds support for AIX Operating system which is now available on Go.

### CODE REVIEW

**Comment by Helflym** [comment]
Any news about this CL ? The failures don't seem related.

File: sigar_aix.go [hunk 1]
```diff
@@ -0,0 +1,538 @@
...
+func bootTime() uint64 {
+	// Get boot time from /etc/utmp
+	file, err := os.Open("/etc/utmp")
+	if err != nil {
+		return 0
+	}
+
+	defer file.Close()
+
+	for {
```
Comment (kvch) [inline sigar_aix.go]: Why is it read in a loop? do we need to wait until the boot time becomes available?

File: sigar_aix.go [hunk 2]
```diff
@@ -0,0 +1,538 @@
...
+		}
+
+		if utmp.Type == C.BOOT_TIME {
+			return uint64(utmp.Time)
+		}
+	}
+	return 0
+}
+
+func tick2msec(val uint64) uint64 {
```
Comment (kvch) [inline sigar_aix.go]: Could you please provide reference for this value as a comment?

File: sigar_aix.go [hunk 3]
```diff
@@ -0,0 +1,538 @@
...
+	//  SBITS is defined in <sys/proc.h>."
+	fixedToFloat64 := func(x uint64) float64 {
+		return float64(x) / (1 << C.SBITS)
+	}
+	self.One = fixedToFloat64(uint64(cpudata.loadavg[0]))
+	self.Five = fixedToFloat64(uint64(cpudata.loadavg[1]))
+	self.Fifteen = fixedToFloat64(uint64(cpudata.loadavg[2]))
+
+	return nil
+
```
Comment (kvch) [inline sigar_aix.go]: Nit: please remove this newline.

File: sigar_aix.go [hunk 4]
```diff
@@ -0,0 +1,538 @@
...
+func (self *Swap) Get() error {
+	ps := C.perfstat_pagingspace_t{}
+	id := C.perfstat_id_t{}
+
+	id.name[0] = 0
+
+	for {
+		if _, err := C.perfstat_pagingspace(&id, &ps, C.sizeof_perfstat_pagingspace_t, 1); err != nil {
+			return nil
```
Comment (kvch) [inline sigar_aix.go]: Why are you ignoring this error?

File: sigar_aix.go (hunk 1 above)
Comment (Helflym) [reply sigar_aix.go]: There is several utmp structures in /etc/utmp. We're looking for one with BOOT_TIME information which might not be the first one.

File: sigar_aix.go (hunk 2 above)
Comment (Helflym) [reply sigar_aix.go]: I've actually explained that sysconf(SC_CLK_TCK) is returning ticks/second in init().

File: sigar_aix.go (hunk 4 above)
Comment (Helflym) [reply sigar_aix.go]: Good question ! That was a mistake I think.

File: sigar_aix.go [hunk 5]
```diff
@@ -0,0 +1,538 @@
...
+	if err != nil {
+		return 0
+	}
+
+	defer file.Close()
+
+	for {
+		var utmp utmp
+		if err := binary.Read(file, binary.BigEndian, &utmp); err != nil {
+			break
```
Comment (fearful-symmetry) [inline sigar_aix.go]: Errors should be reported up the stack.

File: sigar_aix.go [hunk 6]
```diff
@@ -0,0 +1,538 @@
...
+		}
+	}
+	return 0
+}
+
+func tick2msec(val uint64) uint64 {
+	return val * 1000 / system.ticks
+}
+
+func (self *FileSystemList) Get() error {
```
Comment (fearful-symmetry) [inline sigar_aix.go]: Public functions need godocs.

Comment (Helflym) [reply sigar_aix.go]: Hum, there aren't any docs on these functions on others OSes too, and I'm not sure adding something like "Get File System list" would be really useful.

**Comment by fearful-symmetry** [comment]
@Helflym do you want to pick this back up? We're still missing godocs (we might as well add them while the PR is open, and we should standardize the rest of the code) and we should probably get rid of the remaining `_`s in `utmp`

**Comment by fearful-symmetry** [comment]
@Helflym It looks like swap stats require root:

```

**Comment by Helflym** [comment]
> @Helflym It looks like swap stats require root:
> 
> ```
>     TestConcreteGetSwap: concrete_sigar_test.go:58:
>                 Error Trace:    concrete_sigar_test.go:58
>                 Error:          Received unexpected error:
>                                 perfstat_memory_total: permission denied
>                 Test:           TestConcreteGetSwap
> --- FAIL: TestConcreteGetSwap (0.01s)
> ```

No, perfstat API is working without root permissions. The problem is related to how CGO handles errno and perfstat_pagingspace results. I guess errno is being set somewhere during perfstat_pagingspace's call even if it finally succeeds. 
Adding a check on perfstat_pagingspace results should be enough.

**Pull request merged**